cli/command_repository_set_parameters_test.go

```diff
@@ -182,6 +182,71 @@ func (s *formatSpecificTestSuite) TestRepositorySetParametersUpgrade(t *testing.
 	env.RunAndExpectSuccess(t, "index", "epoch", "list")
 }
 
+func (s *formatSpecificTestSuite) TestRepositorySetParametersDowngrade(t *testing.T) {
+	env := s.setupInMemoryRepo(t)
+	out := env.RunAndExpectSuccess(t, "repository", "status")
+
+	// default values
+	require.Contains(t, out, "Max pack length:     21 MB")
+
+	switch s.formatVersion {
+	case format.FormatVersion1:
+		require.Contains(t, out, "Format version:      1")
+		require.Contains(t, out, "Epoch Manager:       disabled")
+		env.RunAndExpectFailure(t, "index", "epoch", "list")
+	case format.FormatVersion2:
+		require.Contains(t, out, "Format version:      2")
+		require.Contains(t, out, "Epoch Manager:       enabled")
+		env.RunAndExpectSuccess(t, "index", "epoch", "list")
+	default:
+		require.Contains(t, out, "Format version:      3")
+		require.Contains(t, out, "Epoch Manager:       enabled")
+		env.RunAndExpectSuccess(t, "index", "epoch", "list")
+	}
+
+	env.Environment["KOPIA_UPGRADE_LOCK_ENABLED"] = "1"
+
+	{
+		cmd := []string{
+			"repository", "upgrade",
+			"--upgrade-owner-id", "owner",
+			"--io-drain-timeout", "1s", "--allow-unsafe-upgrade",
+			"--status-poll-interval", "1s",
+			"--max-permitted-clock-drift", "1s",
+		}
+
+		// You can only upgrade when you are not already upgraded
+		if s.formatVersion < format.MaxFormatVersion {
+			env.RunAndExpectSuccess(t, cmd...)
+		} else {
+			env.RunAndExpectFailure(t, cmd...)
+		}
+	}
+
+	env.RunAndExpectSuccess(t, "repository", "set-parameters", "--upgrade")
```

Comment:
nit; Might be a good idea to write a comment here explaining what is the intention of running the `set-parameters --upgrade` command after running the `upgrade` command above ? I think I know what you might be doing here, but it may not be apparent for the readers.